ddtrace/contrib/asyncio/__init__.py

```diff
@@ -47,8 +47,13 @@
 with require_modules(required_modules) as missing_modules:
     if not missing_modules:
         from .provider import AsyncioContextProvider
+        from ...context import DATADOG_CONTEXT
+        from ...provider import DefaultContextProvider
 
-        context_provider = AsyncioContextProvider()
+        if DATADOG_CONTEXT is None:
```

Comment:
Too bad we couldn't do something like `isinstance(ddtrace.context.ContextManager, ContextVarsContext)`

Or maybe we need a `ddtrace.context.ContextManager.supports_contextvars()`.


I don't really love either of those APIs, but just as random examples.